### ISSUE TITLE
Archive Same Log Multiple Times

### ISSUE THREAD

**Issue by SilvenaRogue** [kauffman12/EQLogParser#171]
In the same vein, not archiving all log files.

Set to 1 Day to archive.  One of my chars log archived 8 times last night.  Then again this morning.  Going back through the folder, this isn't the first time it's happened, just the first time I've noticed.
![Screenshot_5](https://github.com/kauffman12/EQLogParser/assets/164565783/82aaa1ef-e9d1-46f7-b54f-790e033cbb6b)

![multi_archive](https://github.com/kauffman12/EQLogParser/assets/164565783/1dcece36-2c83-4bbe-b277-393fe84541ff)

Other two characters did not archive on opening the parser or on zoning or opening the log files.  One of those two archived mid raid (that was just weird).  So it was a log file that comprised of the previous day and part of a new day when it decided to archive.

![delayed archive](https://github.com/kauffman12/EQLogParser/assets/164565783/49de6fb4-a181-425c-9f69-7908d03d03a2)

Is there a way to do like GINA was doing when opening the program on a new day?  Open GINA, auto archive the log files.  This isn't an issue I noticed previously, because it hadn't happened when I'd been online for a while and zoned multiple times.

Is it based on time of day for the 1 Day to archive?  Like the log file has to be a full 24hrs old before it archives?

Really trying to not be a bother to you.  I love the parser and truly appreciate all the effort you put into this.  You rock.

**Comment by kauffman12** [comment]
Are you using 2.2.18? Also, those characters that it didn't archive. Do you have triggers configured for them? If you turn triggers on/off during a raid it could cause the log files to re-open and an archive to happen. Maybe that needs to be turned off.

The 1 day is supposed to go by the file creation time. I guess you have the size set to any?

**Comment by kauffman12** [comment]
Ok I can probably update STOP to handle the audio better.

 Still not sure why it would try to archive a file that it just did an hour prior though. I'll have to test that and maybe I need to force everything to check in Utc or some standard timezone.

**Comment by SilvenaRogue** [comment]
Truly appreciate the work you do.  Thanks.

**Comment by kauffman12** [comment]
Try 2.2.20. Hopefully it's good now. Otherwise just comment here any re-open.

**Comment by SilvenaRogue** [comment]
Went to update and it still shows 2.2.19.  Also parser set to auto check for updates.

**Comment by kauffman12** [comment]
Try it now. It should be all uploaded.

**Comment by SilvenaRogue** [comment]
This is happening again?  Every time I zone.  I do still have this set to 1 day any size.  
Could we have the option to not archive on zone?  I like the archiving.  So handy.  I just prefer it be daily with size not mattering.  Zoning not having anything to do with it.


This wasn't happening yesterday (7/14) but started up today (7/15)

**Comment by kauffman12** [comment]
yeah that could be an option. just dont understand how it's happening. once a file is archived the replacement should have the new timestamp. it doesn't make much sense. you are using the latest?

**Comment by SilvenaRogue** [comment]
Yup.  Latest version.  I went back in and set it to 3 days versus 1 day for now.  Just so odd.  Logs archived early in the day when I loaded EQLP.  I logged in chars for a bit, zoned a few times on each of them, then logged all but one for a while.  That one archived three more times.  There may have been a zone once on that char, but it archived twice after I logged it out.

The two I loaded in for raid time.  Every time I zoned and what seemed just willy nilly.  Multiples in the same minute, too.

I'm attaching the log.  Redirecting it to parse a fight is what made me notice this.  At one point I did restart EQLP to see if that would clear it up.  Instead it just kept archiving.

It was at 18:00 that I switched it to archive at 3 days vs 1 day.  No more random archiving or archiving on zone.

[EQLogParser-b.txt](https://github.com/user-attachments/files/16243195/EQLogParser-b.txt)

**Comment by kauffman12** [comment]
There's a chance that Windows is caching the date/time. See if this test build is any better. I also have it printing out what it thinks the file creation time is when it archives which might help figure out what's going on.

https://github.com/kauffman12/EQLogParser/raw/master/Release/test/EQLogParser-install-2.2.32-test.exe

**Comment by SilvenaRogue** [comment]
I set everything back to 1 day/Any size last night when I downloaded the test.

There's some error about needing to reload but I'm pretty sure, based on log time, that's when I was logging those characters off anyway.

I spent a bunch of today dealing with UI nonsense, but EQLP archived the log file from yesterday when I relaunched the parser after all that.  Then I zoned a couple of times and opened up the log again.  The log is showing each archived file as having the same creation date and time.

I will say, it's neat seeing log creation time.
[EQLogParser - 7-17.txt](https://github.com/user-attachments/files/16276368/EQLogParser.-.7-17.txt)

**Comment by kauffman12** [comment]
The Created time being printed is in GMT so I'm guessing it was 17:31:10
the day before if you were on the west coast by some chance.
If so then that explains why it would want to do the backup. But the fact
it kept printing the same Created time is pretty bad...

Is it Windows 10 or 11? Uh. I don't know what could make that happen. I can
see if there are other ways to get the file created time.
Maybe I'll make a build that uses local time so it's clearer and prints the
created time using a few different methods to see which one
actually works. At least it looks like it's easy for you to reproduce?

**Comment by SilvenaRogue** [comment]
Yup, west coast.  I'm on Windows 10.

It's pretty easy to reproduce.  At least currently.  Played a different toon to run missions.   Lobby to Inn to mission to Inn and to second mission.  It archived every zone.

**Comment by kauffman12** [comment]
I finally managed to get it to happen for me and none of the options I was hoping to do seem to work. Now in this version I'm trying to manually update the file creation time after archiving it. See if this works. I did run into a problem doing that when the parser and EQ were also using the file but we'll see. Maybe I got lucky the few times I tried it.

https://github.com/kauffman12/EQLogParser/raw/master/Release/test/EQLogParser-install-2.2.32-test2.exe

**Comment by SilvenaRogue** [comment]
> The funny thing is that windows file explorer shows the wrong creation time as well. It's weird.

Right?  I am looking at an empty log file in the EQ log folder (from archiving after I left EQ and relaunched EQLP) from 7/17 21:20 and it has the same creation time of 17:31 PST 7/16.

Very strange.  Haven't fired up the parser or EQ yet.  Will do so when I get a chance and see how this version plays.

**Comment by SilvenaRogue** [comment]
Ok!  I fired up EQLP twice before launching EQ and closed/reloaded it once afterwards.

Initial load it archived the weird empty log files that were leftover with the wrong create date and time.  It left behind new empty log files with accurate create date and time.  Closed and launched again just to see if it'd archive again.  It did not.

Left EQLP running and launched EQ.  No weird archiving.  Zoned a few times, no archiving!!

Closed and relaunched EQLP again.  I had two char logs that never archived (they never zoned yesterday) that were due, I suppose?  Archived with correct date and time and left behind empty log files with accurate create date and time.

Beat up some rabbits and a combat dummy, zoned a few more times.  No extra archiving!

Any idea what in the world was making it wig out?

[EQLogParser 7-18 test.txt](https://github.com/user-attachments/files/16288870/EQLogParser.7-18.test.txt)

**Comment by kauffman12** [comment]
The creation time not updating was the issue. Now I'm setting it myself on
the file after it archives. Shouldn't really need to do that but it's
working.
